router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/local_mock.go

```diff
@@ -0,0 +1,23 @@
+package marketobulkupload
+
+import "net/http"
+
+// mockTransport implements http.RoundTripper interface
+type mockTransport struct {
+	response *http.Response
+	err      error
+}
+
+func (m *mockTransport) RoundTrip(*http.Request) (*http.Response, error) {
+	return m.response, m.err
+}
+
+// createMockClient creates a new http.Client with mocked transport
```

Comment:
We do have a mock roundTripper for OAuth, can we not use that?

Reply:
Will keep a not for backlog for now